### ISSUE TITLE
[Combobox] Spacebar cannot be used to select and deselect options

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3053]
## Current behavior

In expanded Combobox when keyboard-only users get to option they want to select the most common way is to use `spacebar`. But now `spacebar` is used to insert space inside Combobox's input field and it's not selecting or deselecting any option. 

Options can be selected/deselected with `Enter` key but it's not that common for keyboard-only users to interact with options this way.

https://github.com/user-attachments/assets/c1ee5294-2e2f-4358-bc34-a47439bd948f

https://github.com/user-attachments/assets/a0477d6e-7193-49b7-9adf-558a14f1dd13

## Suggested solution

It's recommended to add possibility of selecting option with `spacebar` but also still suport selecting with `Enter` like in example: https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-both/

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS
